packages/hash/frontend/src/components/BlockLoader/data-map-editor.tsx

```diff
@@ -0,0 +1,357 @@
+ *    - @todo store maps in a database
+ * 3. Proper exploration as to how this works in practice and what additional functionality is required.
+ */
+export const DataMapEditor = ({
+  close,
+  mapId,
+  schemaMap,
+  sourceBlockEntity,
+  sourceBlockGraph,
+  targetSchema,
+  transformedTree,
+  updateSchemaMap,
+}: DataMapperProps): ReactElement => {
+  const [showPreview, setShowPreview] = useState(false);
+
+  const { sourcePaths, sourceTree } = useMemo(() => {
+    const tree = dataTreeFromEntityGraph(sourceBlockEntity, sourceBlockGraph);
+    const paths: string[] = derivePathsFromObject(tree);
+    return {
+      sourceTree: tree,
+      sourcePaths: paths,
+    };
+  }, [sourceBlockEntity, sourceBlockGraph]);
+
+  const targetPaths = useMemo(
+    () => derivePathsFromSchema(targetSchema, targetSchema.properties),
+    [targetSchema],
+  );
+
+  if (!schemaMap?.transformations) {
+    const initialTransformations = generateInitialTransformations(
+      sourcePaths,
+      targetPaths,
+    );
+    updateSchemaMap((map) => ({
+      ...map,
+      transformations: initialTransformations ?? {},
+    }));
+  }
+
+  return (
+    <Box sx={{ position: "relative" }}>
+      <IconButton
+        sx={{
+          position: "absolute",
+          top: 8,
+          right: 0,
+        }}
+        onClick={() => close()}
+        type="button"
+      >
+        <FontAwesomeIcon
+          icon={faClose}
+          sx={{
+            color: "gray.50",
+            fontSize: 24,
+          }}
+        />
+      </IconButton>
+      <Box>
+        <Box
+          sx={({ palette }) => ({
+            display: "flex",
+            borderBottom: `1px solid ${palette.gray[30]}`,
+            py: 1,
+          })}
+        >
+          <Box width="30%">
+            <Typography fontWeight={600} variant="smallTextLabels">
+              Target path
+            </Typography>
+          </Box>
+          <Box width="30%">
+            <Typography fontWeight={600} variant="smallTextLabels">
+              Source path
+            </Typography>
+          </Box>
+          <Box width="30%" ml={4}>
+            <Typography fontWeight={600} variant="smallTextLabels">
+              Default
+            </Typography>
+          </Box>
+        </Box>
+
+        {targetPaths.map((targetPath) => {
+          const { required, schema: targetPathSchema } = schemaByPathParts(
+            targetSchema,
+            targetPath.split("."),
+          );
+
+          const { description, type } = targetPathSchema;
+
+          return (
+            <Box
+              key={targetPath}
+              sx={({ palette }) => ({
+                py: 1,
+                display: "flex",
+                alignItems: "center",
+                borderBottom: `1px solid ${palette.gray[20]}`,
+              })}
+            >
+              <Box width="30%">
```

Comment:
Can you use flex rather than `width` here? If all of these have `flex: 1` I think this will work itself out.

Reply:
Done in https://github.com/hashintel/hash/pull/918/commits/5080167c62a147a2dc87c0c9bdb0bc9be505001b